deployer/update_grafana_datasources.py

```diff
@@ -143,31 +139,54 @@ def build_request_headers():
     return headers
 
 
-def get_clusters_used_as_datasources():
+def get_clusters_used_as_datasources(cluster_name, datasource_endpoint):
     """Returns a list of cluster names that have prometheus instances already defined as datasources of the centralized Grafana."""
-    headers = build_request_headers()
+    headers = build_request_headers(cluster_name)
     # Get a list of all the currently existing datasources
-    response = requests.get(DATASOURCE_ENDPOINT, headers=headers)
+    response = requests.get(datasource_endpoint, headers=headers)
 
     if response.status_code != 200:
         print(
-            f"An error occured when retrieving the datasources from {DATASOURCE_ENDPOINT}. \n Error was {response.text}."
+            f"An error occured when retrieving the datasources from {datasource_endpoint}. \n Error was {response.text}."
         )
         response.raise_for_status()
 
-    print("Successfully retrieved the datasources!")
     datasources = response.json()
-    return [datasource["name"] for datasource in datasources]
+    datasources = [datasource["name"] for datasource in datasources]
+    print_colour(
+        f"Successfully retrieved {len(datasources)} existing datasources! {datasources}"
+    )
+
+    return datasources
 
 
 def main():
+    argparser = argparse.ArgumentParser(
+        description="""A command line tool to update Grafana
+        datasources.
+        """
+    )
+
+    argparser.add_argument(
```

Comment:
set the defaults here to be the 2i2c cluster's grafana?

Reply:
I did forget to set defaults, but did it in a later commit https://github.com/2i2c-org/infrastructure/pull/1215/commits/2e1e3c3d05ddf1abafaccfd28bf994c6f9a43231